src/concrete/ml/sklearn/__init__.py

```diff
@@ -26,100 +30,173 @@ def get_sklearn_models():
 
     Args:
-        prelist: list of models
-        classifier (bool): whether you want classifiers or not
-        regressor (bool): whether you want regressors or not
-        str_in_class_name (List[str]): if not None, only return models with the given string or
-            list of strings as a substring in their class name
+        models: The list of models to consider.
+        classifier (bool): If classifiers should be considered.
+        regressor (bool): If regressors should be considered.
+        include (Optional[Union[str, List[str]]]): If not None, only return models which names (or
+            a part of it) match the given string or list of strings. Default to None.
+        exclude (Optional[Union[str, List[str]]]): If not None, only return models which names (or
+            a part of it) do not match the given string or list of strings. Default to None.
 
     Returns:
-        the sublist which fulfills the (classifier, regressor, str_in_class_name) conditions.
+        The filtered list of models available in Concrete ML, sorted by name.
 
     """
     assert_true(classifier or regressor, "Please set at least one option")
 
-    answer = []
+    selected_models = []
 
     if classifier:
-        answer += [m for m in prelist if is_classifier_or_partial_classifier(m)]
+        selected_models += [model for model in models if is_classifier_or_partial_classifier(model)]
 
     if regressor:
-        answer += [m for m in prelist if is_regressor_or_partial_regressor(m)]
+        selected_models += [model for model in models if is_regressor_or_partial_regressor(model)]
 
-    if str_in_class_name is not None:
-        if isinstance(str_in_class_name, str):
-            str_in_class_name = [str_in_class_name]
+    if include is not None:
+        if isinstance(include, str):
+            include = [include]
 
-        for name in str_in_class_name:
-            answer += [m for m in answer if name in m.__name__]
+        for name in include:
+            selected_models = [model for model in selected_models if name in get_model_name(model)]
 
-    # We return a sorted list such that it is ordered, to avoid notably issues when it is used
-    # in @pytest.mark.parametrize
-    return sorted(answer, key=lambda m: m.__name__)
+    if exclude is not None:
+        if isinstance(exclude, str):
+            exclude = [exclude]
+
+        for name in exclude:
+            selected_models = [
+                model for model in selected_models if name not in get_model_name(model)
+            ]
+
+    # Return a sorted list in order to avoid issues when used in @pytest.mark.parametrize
+    return sorted(selected_models, key=lambda m: m.__name__)
 
 
 def get_sklearn_linear_models(
-    classifier: bool = True, regressor: bool = True, str_in_class_name: List[str] = None
+    classifier: bool = True,
+    regressor: bool = True,
+    str_in_class_name: Optional[Union[str, List[str]]] = None,
```

Comment:
I would have preferred to put `include` and `exclude` (as I've done for the new or private functions) but I did not because that would make this an API breaking change (+ it's not only used in our test suite)

Reply:
Do people really use these methods?

Maybe we should be more clear between what's private and what's not because I honestly don't see any user using these methods tbh

Reply:
> really? where is it used elsewhere?

in our serialization's decoder, because we need to provide a list of supported/accepted models from CML !

I kinda of agree with you @fd0r and would say it's fine if I make this change, but I'd prefer to be sure if it's fine with you first !

Reply:
Fine with me to break this API without changing a major.
But it would probably be best to use the `_` prefix to signal to the user that the function is private.

They can still use if they want to do but then we properly signal that this API can change.

Reply:
ok got it, I'll make them private then !

Reply:
done